sphinxcontrib/needs/directives/need.py

```diff
@@ -107,9 +123,9 @@ def run(self):
         )
         return need_nodes
 
-    def read_in_links(self, name):
+    def read_in_links(self, name: str) -> List[str]:
         # Get links
-        links_string = self.options.get(name, [])
+        links_string = self.options.get(name)
```

Comment:
MMh wait, why does this work?
Old code was ``links_string = self.options.get(name, [])`` 

For me, it makes sense to have a default value here because as far as I know not all options are available if not set in the directive.
Same thoughts for all the `self.options.get()` calls above?

Is this change related to the typing annotations or is it generally not needed?

Reply:
`dict.get` has a default default value of `None`, so it's not necessary to set a value here unless `None` is not appropriate.
i removed the default list because it doesn't make sense (from a type-safety perspective) to assign an empty list to a variable which would otherwise be used to represent a `str`